sweep/test_utils.go

```diff
@@ -155,12 +155,24 @@ func (m *MockNotifier) RegisterBlockEpochNtfn(
 	log.Tracef("Mock block ntfn registered")
 
 	m.mutex.Lock()
-	epochChan := make(chan *chainntnfs.BlockEpoch, 0)
-	bestHeight := int32(0)
-	if bestBlock != nil {
-		bestHeight = bestBlock.Height
+
+	// The real notifier returns a notification with the current block hash
+	// and height immediately if no best block hash or height is specified
+	// in the request. We want to emulate this behaviour as well for the
+	// mock.
+	var epochChan chan *chainntnfs.BlockEpoch
+	switch {
+	case bestBlock == nil:
+		epochChan = make(chan *chainntnfs.BlockEpoch, 1)
+		epochChan <- &chainntnfs.BlockEpoch{
+			Hash:   mockChainHash,
+			Height: mockChainHeight,
+		}
+		m.epochChan[epochChan] = mockChainHeight
+	default:
+		epochChan = make(chan *chainntnfs.BlockEpoch)
+		m.epochChan[epochChan] = bestBlock.Height
```

Comment:
Not sure if the two different channel buffer sizes are going to create surprises for future devs? Maybe this initial send can be done in a goroutine. Or will that create the risk that the first block that is manually pushed in comes before that initial send? Could be prevented by having a notifyBlock method that first checks whether the initial send goroutine has completed.

Reply:
It seems to work fine with a goroutine. Just didn't think about doing it that way when implementing it.
But makes a lot of sense to do it that way to not create surprises for future devs.